src/libraries/Common/src/System/IO/PathInternal.cs

```diff
@@ -241,6 +241,12 @@ internal static bool RemoveRelativeSegments(ReadOnlySpan<char> path, int rootLen
         internal static bool EndsInDirectorySeparator(ReadOnlySpan<char> path) =>
             path.Length > 0 && IsDirectorySeparator(path[path.Length - 1]);
 
+        /// <summary>
+        /// Same as <see cref="EndsInDirectorySeparator(string?)"/> but without length check.
+        /// </summary>
+        internal static bool EndsInDirectorySeparatorUnchecked(ReadOnlySpan<char> path) =>
```

Comment:
I suggest to just inline both of EndsInDirectorySeparatorUnchecked and EndsInDirectorySeparator since they are so simple and only exist in 3 or 4 places. then you can remove these methods.